query-service-impl/src/main/java/org/hypertrace/core/query/service/HandlerScopedMaskingConfig.java

```diff
@@ -0,0 +1,158 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class HandlerScopedMaskingConfig {
+  private static final String TENANT_SCOPED_MASKS_CONFIG_KEY = "tenantScopedMaskingCriteria";
+  private final Map<String, List<MaskValuesForTimeRange>> tenantToMaskValuesMap;
+  private HashMap<String, Boolean> shouldMaskAttribute = new HashMap<>();
+  private HashMap<String, String> maskedValue = new HashMap<>();
+
+  public HandlerScopedMaskingConfig(Config config) {
+    if (config.hasPath(TENANT_SCOPED_MASKS_CONFIG_KEY)) {
+      this.tenantToMaskValuesMap =
+          config.getConfigList(TENANT_SCOPED_MASKS_CONFIG_KEY).stream()
+              .map(maskConfig -> new TenantMasks(maskConfig))
+              .collect(
+                  Collectors.toMap(
+                      tenantFilters -> tenantFilters.tenantId,
+                      tenantFilters -> tenantFilters.maskValues));
+    } else {
+      this.tenantToMaskValuesMap = Collections.emptyMap();
+    }
+  }
+
+  public void parseColumns(ExecutionContext executionContext) {
+    shouldMaskAttribute.clear();
+    String tenantId = executionContext.getTenantId();
+
+    if (!tenantToMaskValuesMap.containsKey(tenantId)) {
+      return;
+    }
+
+    Optional<QueryTimeRange> queryTimeRange = executionContext.getQueryTimeRange();
+    Instant queryStartTime, queryEndTime;
+    if (queryTimeRange.isPresent()) {
+      queryStartTime = queryTimeRange.get().getStartTime();
+      queryEndTime = queryTimeRange.get().getEndTime();
+    } else {
+      queryEndTime = Instant.MAX;
+      queryStartTime = Instant.MIN;
+    }
+    for (MaskValuesForTimeRange timeRangeAndMasks : tenantToMaskValuesMap.get(tenantId)) {
+      boolean timeRangeOverlap =
+          isTimeRangeOverlap(timeRangeAndMasks, queryStartTime, queryEndTime);
+
+      if (timeRangeOverlap) {
+        Map<String, String> attributeToMaskedValue =
+            timeRangeAndMasks.maskValues.attributeToMaskedValue;
+        for (String attribute : attributeToMaskedValue.keySet()) {
+          shouldMaskAttribute.put(attribute, true);
+          maskedValue.put(attribute, attributeToMaskedValue.get(attribute));
+        }
+      }
+    }
+  }
+
+  private static boolean isTimeRangeOverlap(
+      MaskValuesForTimeRange timeRangeAndMasks, Instant queryStartTime, Instant queryEndTime) {
+    boolean timeRangeOverlap = true;
+
+    if (timeRangeAndMasks.getStartTimeMillis().isPresent()) {
+      Instant startTimeInstant = Instant.ofEpochMilli(timeRangeAndMasks.getStartTimeMillis().get());
+      if (startTimeInstant.isBefore(queryStartTime) || startTimeInstant.isAfter(queryEndTime)) {
+        timeRangeOverlap = false;
+      }
+
+      Instant endTimeInstant = Instant.ofEpochMilli(timeRangeAndMasks.getStartTimeMillis().get());
```

Comment:
should be `getEndTimeMillis`?

Reply:
I guess, is this what we are looking as function?
```
if (timeRangeAndMasks.getStartTimeMillis().isPresent()) {
      
      Instant startTimeInstant = Instant.ofEpochMilli(timeRangeAndMasks.getStartTimeMillis().get());
      Instant endTimeInstant = Instant.ofEpochMilli(timeRangeAndMasks.getEndTimeMillis().get());

      if (!(startTimeInstant.isAfter(queryEndTime) || endTimeInstant.isBefore(queryStartTime))) {
        return true;
      }
    }
    return false;
```

Reply:
Right, fixed the condition.